workloads/amd64/rke2-upgrade-plan.yaml

```diff
@@ -30,20 +30,34 @@ spec:
   version: $UPGRADEVERSION
   nodeSelector:
     matchExpressions:
-        - key: "rke.cattle.io/etcd-role"
-          operator: Exists
-        - key: node-role.kubernetes.io/etcd
-          operator: In
-          values: [ "true" ]
-        - key: node-role.kubernetes.io/control-plane
-          operator: NotIn
-          values: [ "true" ]
+      - {key: node-role.kubernetes.io/etcd, operator: In, values: ["true"]}
+      - {key: node-role.kubernetes.io/control-plane, operator: NotIn, values: ["true"]}
   tolerations:
     - operator: Exists
   serviceAccountName: system-upgrade
-  prepare:
+  cordon: true
+  drain:
+    force: true
+  upgrade:
     image: rancher/rke2-upgrade
-    args: ["prepare", "rke2-server-cp"]
```

Comment:
why did you remove the `prepare` bit here? Without this, none of the plans will wait for other plans to finish before upgrading. They will all just upgrade at once.

you need to have `args: ["prepare", "PLAN-TO-WAIT-FOR"]` in order for one plan to wait for another to complete before proceeding.